ynput/ayon-core
# Chore: Applications addon

## Changelog Description
Move applications logic from lib to addon. And modified related code accordingly.

## Additional info
Applications addon does contain everything what `applications.py` in lib did. Moved `extractenvironments` command to the addon as it is his responsibility to implement it. Moved collect plugin which cares about collection of app name/label, or host name based on application name. Modified deadline and royalrender to support both options of arguments. Marked global command `extractenvironments` as deprecated.

Core changes: Created folder `addons` inside core addon which can be used for converted addons that are not yet in different repository. The first subfolder is name of an addon which is added to `sys.path`. So any content in the folder will be added to sys path and can be imported as with addons client code. This approach was chosen as future prove approach, to be able to skip addons which already are separated into repository, but older version of ayon core is used. This approach does not add the addons to PYTHONPATH so it may have issues with host addons as they might not be able to just import their code directly.

Opened questions:
- Should global prelaunch hooks be in applications addon or in core addon?
- Should we split prelaunch hooks that are specific for limited amount of hosts into the hosts?

## Testing notes:
There is no visible change, all should work as before.
1. For farm publishing update plugins accordingly.
2. Launch application of your choice.
3. Run publishing.
4. Run publishing through farm.
5. All should work.

PR partially collides with changes in https://github.com/ynput/ayon-core/pull/197 . The PR will be modified accordingly after this PR merge.
